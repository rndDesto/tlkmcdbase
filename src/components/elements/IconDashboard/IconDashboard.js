import React from 'react';
import PropTypes from 'prop-types';
import { autoPx } from '../../../utils/unit';

export default function IconDashboard({ fill }) {
  const r24 = autoPx(24);
  const d1 = `M12.5953 5H19V9.45437H12.5953V5Z`;
  const d2 = `M5 13.3725H11.4047V5.001H5V13.3725Z`;
  const d3 = `M12.5953 10.6295H19V19H12.5953V10.6295Z`;
  const d4 = `M5 14.5456H11.4047V19H5V14.5456Z`;

  return (
    <svg fill="none" height={r24} viewBox="0 0 24 24" width={r24} xmlns="http://www.w3.org/2000/svg">
      <path d={d1} fill={fill} />
      <path d={d2} fill={fill} />
      <path d={d3} fill={fill} />
      <path d={d4} fill={fill} />
    </svg>
  );
}

IconDashboard.defaultProps = {
  fill: '#25282B',
};

IconDashboard.propTypes = {
  fill: PropTypes.string,
};
