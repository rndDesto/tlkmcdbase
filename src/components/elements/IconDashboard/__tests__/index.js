import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import IconDashboard from '../IconDashboard';

jest.mock('../../../../utils/unit', () => ({
  autoPx: v => v,
}));

describe('src/components/elements/IconDashboard', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<IconDashboard />);
    expect(tree).toMatchSnapshot();
  });
});
