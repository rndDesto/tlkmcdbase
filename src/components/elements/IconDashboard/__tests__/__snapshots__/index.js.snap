// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/IconDashboard render 1`] = `
<svg
  fill="none"
  height={24}
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12.5953 5H19V9.45437H12.5953V5Z"
    fill="#25282B"
  />
  <path
    d="M5 13.3725H11.4047V5.001H5V13.3725Z"
    fill="#25282B"
  />
  <path
    d="M12.5953 10.6295H19V19H12.5953V10.6295Z"
    fill="#25282B"
  />
  <path
    d="M5 14.5456H11.4047V19H5V14.5456Z"
    fill="#25282B"
  />
</svg>
`;
